src/main/java/com/fasterxml/jackson/databind/node/DecimalNode.java

```diff
@@ -126,11 +126,23 @@ public boolean equals(Object o)
         if (o == this) return true;
         if (o == null) return false;
         if (o instanceof DecimalNode) {
-            return ((DecimalNode) o)._value.compareTo(_value) == 0;
+            DecimalNode otherNode = (DecimalNode) o;
+            if (otherNode._value == null) {
+                return _value == null;
+            } else if (_value == null) {
+                return false;
+            }
+            return otherNode._value.compareTo(_value) == 0;
         }
         return false;
     }
 
     @Override
-    public int hashCode() { return Double.valueOf(doubleValue()).hashCode(); }
+    public int hashCode() {
+        if (_value == null) {
+            // we need a stable hash code for _value == null
+            return 0;
+        }
+        return Double.valueOf(doubleValue()).hashCode();
```

Comment:
huh? Oh, that's how code already was. Ok. :)

Reply:
But not sure why not use `BigDecimal.hashCode()`, fwtw.
Maybe something else to change in.... 2.17 I think.

Reply:
The pre-existing equals method is odd as well. It would be easier to use `.equals` on the BigDecimal instances.

Reply:
compareTo and the special hashCode are necessary here because BigDecimal equality is weird wrt scaling. 1.9 and 1.90 are considered unequal by BigDecimal but equal by this DecimalNode impl.

Reply:
Ok; will then leave that as-is.